internal/service/advancedcluster/resource_advanced_cluster.go

```diff
@@ -46,6 +46,11 @@ const (
 
 var DeprecationMsgOldSchema = fmt.Sprintf("%s %s", constant.DeprecationParam, DeprecationOldSchemaAction)
 
+type OldShardConfigMeta struct {
```

Comment:
I would move this to model_advanced_cluster

Reply:
Done